### DISCUSSION TITLE
getting "Could not boot: Error 0x7f048283 (https://ipxe.org/7f048283)" when booting fedora 35 vmlinuz/initrd

### DISCUSSION THREAD

**Discussion by hradec** [ipxe/ipxe#714] [Q&A]
I'm building ipxe from the latest master branch to boot over the network using bios legacy and efi network boot. (I'm serving the ipxe bios/efi payloads to the network PXE boot rom via the standard DHCP/TFTP combo, using dnsmasq)
I'm adding a custom script to ipxe that shows a menu to select booting our linux distro over the network or to the local disk boot. 

The script works fine when booting on a regular bios network boot, but when booting on a efi network boot, I get this `Could not boot: Error 0x7f048283 (https://ipxe.org/7f048283)` error message after ipxe loads the vmlinuz kernel and the initrd image from our tftp: 
```
>>Start PXE over IPv4.
  Station IP address is 192.168.1.80

  Server IP address is 192.168.1.231
  NBP filename is payload.efi
  NBP filesize is 1017344 Bytes
 Downloading NBP file...

  NBP file downloaded successfully.
iPXE initialising devices...ok

iPXE 1.21.1+ (gc5af4) -- Open Source Network Boot Firmware -- https://ipxe.org
Features: DNS HTTP iSCSI TFTP SRP AoE EFI Menu
Configuring (net0 e2:c0:ed:fd:42:19).....

...

tftp://192.168.1.231/vmlinuz... ok 
tftp://192.168.1.231/initrd... ok 
Could not boot: Error 0x7f048283 (https://ipxe.org/7f048283)
```
I have tried building with `DEBUG=efi_image`, but that gives me the same error message twice, in different colors. 

I'm currently able to boot the same vmlinuz/initrd on EFI network boot computers/VMs using GRUB2 instead of ipxe, so I don't think it's a problem with the vmlinuz kernel file. 

To sum up: 

- vmlinuz boots fine from ipxe on bios network boot
- ipxe can't boot vmlinuz on efi network boot, and spits the 0x7f048283 error
- grub2 is able to boot vmlinuz on efi network boot just fine. 

this is the custom script.ipxe I'm using when building booth bios and efi ipxe payloads: 
```
#!ipxe

:retry_dhcp
dhcp || goto retry_dhcp

:start
menu iPXE boot menu
item --gap --
item Fedora.35  Fedora.35
item Local.Windows      Local.Windows
choose --timeout 10000 --default Local.Windows selected
set menu-timeout 0
goto ${selected}

:Fedora.35
kernel tftp://192.168.1.231/vmlinuz
initrd tftp://192.168.1.231/initrd
imgargs vmlinuz  quiet nb_provisionurl=tftp://192.168.1.231/netbootcd-ipxe-bootchain/boot/init.sh  nofstab waitusb norestore base console=tty0  console=ttyS0,115200n8
boot || sleep 600
goto start

:Local.Windows
goto exit_ipxe

:exit_ipxe
 ```
Is there something different that I have to do for the efi script maybe?

**Comment by NiKiZe** [comment]
By following https://ipxe.org/7f048283 You get a "see also" that goes to https://ipxe.org/err/7f048200
> Your EFI firmware could not start the provided binary, make sure there is no corruption while downloading image for example by checking filesize with imgstat, also check that the image on the server is not corrupt. Another thing to make sure is that Secure Boot is disabled.

Is secure boot enabled? When you are using grub you probably have shim.efi, grub uses a different method to execute the kernel, while iPXE leaves the verification and execution to EFI firmware. The kernel is often signed with a key only valid in grub/shim. Not for firmware booting.

**Reply by stappersg** [reply]
ping @hradec

**Reply by stappersg** [reply]
FWIW https://github.com/ipxe/ipxe/issues/762 is also opened by @hradec
